### PR TITLE
User authentication/authorisation and backend for comments

### PR DESCRIPTION
**Pull request by shazzatherazza**
Addresses issue #52 completely, and addresses the backend but not the frontend portions of issues #7 and #6.

### CODE REVIEW

File: backend/app.js [hunk 1]
```diff
@@ -25,11 +27,27 @@ app.use(cookieParser())
 app.use(express.static(path.join(__dirname, 'public')))
 app.use(cors())
 
+// use sessions for tracking logins
+app.use(session({
+  secret: 'home2health secret key',
```
Comment (noobling) [inline backend/app.js]: this is okay for now but this is a really sensitive part of your app that should be in .env

If other people get hold of this they can create any cookie they want and be authenticated as anyone

File: backend/models/user.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+// authenticate input against database documents
+UserSchema.statics.authenticate = function(email, password, callback) {
+    User.findOne({email: email})
+        .exec(function (error, user) {
+            if (error) {
+                return callback(error);
+            }
+            else if (!user) {
+                var err = new Error('User not found.');
+                err.status = 401;
```
Comment (noobling) [inline backend/models/user.js]: You probably want a more semantically correct error code like 400 bad requests or 404 not found

File: backend/routes/post.js [hunk 3]
```diff
@@ -46,53 +51,107 @@ router.get('/', (request, response, next) => {
...
     .deletePost(currentPostId)
     .then(data => {
-      response.send(data)
+      User.findById(request.session.userId, function(err, doc) {
+        if (err) return next(err)
+        if (!doc) {
+          err = new Error('User Not Found')
+          err.status = 404
+          return next(err)
+        }
```
Comment (noobling) [inline backend/routes/post.js]: This code will be repeated a lot you could move it to a function or middleware

**Comment by noobling** [comment]
@shazzatherazza this PR is no longer relevant is it?

**Comment by shazzatherazza** [comment]
> @shazzatherazza this PR is no longer relevant is it?

Nope. I'll close it.

**Pull request closed without merging**